testdata/testoutput5

```diff
@@ -1972,6 +1972,20 @@ No match
     \x{09f}
 No match
 
+/^\p{Cs}/match_invalid_utf
+\= Expect no match
+    \x{dfff}\=no_utf_check
+No match
+    \x{09f}
+No match
+
+/^\p{Cs}/match_invalid_utf,allow_surrogates
```

Comment:
this replicates the behaviour where utf is used instead of match_invalid_utf, by using the "allow_surrogates" to redefine what is valid, but maybe it shouldn't match instead since it is invalid UTF-16?